majilove/models.py

```diff
@@ -106,3 +108,402 @@ def display_blood_type(self):
+            'english': 'Stamina based',
+            # unsure about this one too
+            'japanese_translation': u'LIFEでクリア時',
+
+            'variables': ['percentage', 'amount'],
+            'template': _(u'+{amount} score when clearing a song with {percentage}% Stamina'),
+            'japanese_template': u'LIFE{percentage}%以上でクリア時+{amount}スコア'
+        })
+    ])
+
+    ALL_SUB_SKILL_VARIABLES = { item: True for sublist in [ _info['variables'] for _info in SUB_SKILL_TYPES.values()] for item in sublist }.keys()
+
+    SUB_SKILL_TYPE_CHOICES = [(_name, _info['translation']) for _name, _info in SUB_SKILL_TYPES.items()]
+    SUB_SKILL_TYPE_WITHOUT_I_CHOICES = True
+    i_sub_skill_type = models.PositiveIntegerField(_('Sub Skill'), choices=SUB_SKILL_TYPE_CHOICES, null=True)
+    japanese_sub_skill_type = property(getInfoFromChoices('sub_skill_type', SUB_SKILL_TYPES, 'japanese_translation'))
+
+    @property
+    def sub_skill_template(self):
+        return self.SUB_SKILL_TYPES[self.sub_skill_type]['template']
+
+    @property
+    def japanese_sub_skill_template(self):
+        return self.SUB_SKILL_TYPES[self.sub_skill_type]['japanese_template']
+
+    @property
+    def sub_skill_variables(self):
+        return {
+            key: getattr(self, u'sub_skill_{}'.format(key))
+            for key in self.SUB_SKILL_TYPES[self.sub_skill_type]['variables']
+        }
+
+    @property
+    def sub_skill(self):
+        if self.i_sub_skill_type is None: return None
+        return self.sub_skill_template.format(**self.sub_skill_variables)
+
+    @property
+    def japanese_sub_skill(self):
+        if self.i_sub_skill_type is None: return None
+        return self.japanese_sub_skill_template.format(**self.sub_skill_variables)
+
+    sub_skill_amount = models.PositiveIntegerField('{amount}', null=True)
+    sub_skill_percentage = models.FloatField('{percentage}', null=True)
+
+    # Cache idol
+
+    _cache_idol_days = 20
+    _cache_idol_last_update = models.DateTimeField(null=True)
+    _cache_j_idol = models.TextField(null=True)
+
+    @classmethod
+    def cached_idol_pre(self, d):
+        d['name'] = d['names'].get('en', None)
+        d['t_name'] = d['unicode'] = d['names'].get(get_language(), d['name'])
+
+    def to_cache_member(self):
+        if not self.idol:
+            return {
+                'id': None,
+                'names': {},
+                'image': None,
+            }
+        names = self.idol.names or {}
+        names['en'] = self.idol.name
+        names['ja'] = self.idol.japanese_name
+        return {
+            'id': self.idol.id,
+            'names': names,
+            'image': unicode(self.idol.image)
+        }
+
+    def __unicode__(self):
+        if self.id:
+            return u'{rarity} {idol_name} - {name}'.format(
+                rarity = self.rarity,
+                idol_name = self.cached_idol.t_name if self.cached_idol else '',
+                name=(self.japanese_name
```

Comment:
Since there's no reason to store `japanese_name` separately (see my previous comment), you can just do:

```python
name=self.t_name
```